codemod_yaml/box/yaml/sequence.py

```diff
@@ -49,17 +74,21 @@ def __setitem__(self, index: int, other: Any) -> None:
             self.stream.edit(t, other)
         self._items[index].value = other
 
-    def __delitem__(self, index: int) -> None:
-        self._ensure(index)
+    @overload
+    def __delitem__(self, index: slice) -> None: ...
+    @overload
+    def __delitem__(self, index: int) -> None: ...
+    def __delitem__(self, index):  # type: ignore[no-untyped-def]
```

Comment:
Wow. What syntax is this?

Reply:
It's part of the typing pep, https://peps.python.org/pep-0484/#function-method-overloading

It's very common in typeshed's `.pyi` files, but very uncommon in user code.  Note the pep has this gem, which doesn't appear to be true in mypy:

> the non-@overload-decorated definition [is] used at runtime but should be ignored by a type checker.

The `# type: ignore[no-untyped-def]` appears to be required for me.